src/messages/MessageBuilder.cpp

```diff
@@ -2746,6 +2757,28 @@ void MessageBuilder::appendTwitchBadges(const QVariantMap &tags,
         return;
     }
 
+    if (this->message().flags.has(MessageFlag::SharedMessage))
+    {
+        QString sourceId = tags["source-room-id"].toString();
+        std::optional<QString> sourceName;
+        if (twitchChannel->roomId() == sourceId)
+        {
+            sourceName = twitchChannel->getName();
```

Comment:
This assumes an incoming message to channel A is a shared message where the `source-room-id` is also channel A - can this actually happen?

Reply:
due to the latest refactor, yeah `twitchChannel` in that method can be the `sourceChannel` (field no longer exists) rather than the TwitchChannel that received the message: https://github.com/Chatterino/chatterino2/blob/e35fabfabe2f430e5614340a0fc6e5009db415b3/src/messages/MessageBuilder.cpp#L1946